### PR TITLE
fix(wallet): fix ctrl-c handler on windows platflorm

### LINKED ISSUE #1598: [Wallet] Immediately closes after starting up on Windows

### PR DESCRIPTION
**Pull request by tmpolaczyk**
Fix #1598

### CODE REVIEW

File: wallet/src/signal.rs [hunk 1]
```diff
@@ -5,14 +5,19 @@ use actix::prelude::*;
 pub fn ctrl_c<T: Fn() + 'static>(cb: T) {
     #[cfg(unix)]
     let sigterm = tokio_signal::unix::Signal::new(tokio_signal::unix::SIGTERM)
-        .map(|s| s.map(|_| ()))
+        .map(|s| {
+            log::trace!("Received SIGTERM signal");
```
Comment (tmpolaczyk) [inline wallet/src/signal.rs:9]: This is received when doing `kill $(pidof witnet)`, removing this handler will make the wallet shutdown instantly and corrupt the database.

File: wallet/src/signal.rs [hunk 2]
```diff
@@ -5,14 +5,19 @@ use actix::prelude::*;
...
         .flatten_stream();
 
+    // There is no equivalent to SIGTERM on Windows, so use empty stream
     #[cfg(windows)]
-    let sigterm = futures::future::ok(()).into_stream();
+    let sigterm = futures::stream::empty();
 
     let ctrl_c = tokio_signal::ctrl_c().flatten_stream();
     let handle_shutdown = ctrl_c
+        .inspect(|_| log::trace!("Received CTRL-C"))
```
Comment (tmpolaczyk) [inline wallet/src/signal.rs:20]: This is received when pressing CTRL-C, but not sure if it works on windows.

File: wallet/src/signal.rs [hunk 3]
```diff
@@ -5,14 +5,16 @@ use actix::prelude::*;
 pub fn ctrl_c<T: Fn() + 'static>(cb: T) {
     #[cfg(unix)]
     let sigterm = tokio_signal::unix::Signal::new(tokio_signal::unix::SIGTERM)
-        .map(|s| s.map(|_| ()))
+        .map(|s| s.map(|_| log::trace!("Received SIGTERM signal")))
```
Comment (tmpolaczyk) [inline wallet/src/signal.rs:8]: This is received when doing `kill $(pidof witnet)`, removing this handler will make the wallet shutdown instantly and corrupt the database.

Comment (tmpolaczyk) [reply wallet/src/signal.rs:10]: I decided to put this comments into the code

**Review by aesedepece — APPROVED** [review]

**Review by mariocao — APPROVED** [review]

**Review by girazoki — APPROVED** [review]

**Comment by mariocao** [comment]
Closed in 81ae26849fea6de533d3b15ce9f19dffe2359810

**Pull request closed without merging**